services/explorer/src/components/Header/Header.tsx

```diff
@@ -8,25 +8,42 @@ import {
 } from '@taraxa_project/taraxa-ui';
 import { Box, IconButton } from '@mui/material';
 import MenuIcon from '@mui/icons-material/Menu';
+import { theme } from '../../theme-provider';
 import { TaraxaIcon } from '../icons';
 import { HeaderBtn, useHeaderEffects } from './Header.effects';
 
 export const Header = () => {
   const isMobile = useMediaQuery({ query: `(max-width: 760px)` });
-  const { headerButtons } = useHeaderEffects();
+  const { headerButtons, selected } = useHeaderEffects();
+  const buttonRow = selected
```

Comment:
This seems a bit overkill to me.
Why can't we keep the whole array of headerButtons in the state and when a location changes, we just set the state and update the current header btn as selected. Then here we don't need to iterate again and modify the property on a button.

Reply:
Right, I'll refactor